### PR TITLE
 Oliinyk / 2093 Possibility to open workshop recruitment when there is no seats for

### PR DESCRIPTION
**Pull request by doliinyk**
#2093 

Add modal for workshop seats lack case that offers to change seats number; also provided min seats number in workshop editing if seats are already taken or there are recruitment open (taken seats + 1).

### CODE REVIEW

File: src/app/shared/components/workshop-card/workshop-seats-lack-modal/workshop-seats-lack-modal.component.html [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  </p>
+
+  <mat-dialog-actions fxLayout="row wrap" fxLayoutAlign="center center">
+    <a
+      class="btn"
+      mat-button
+      cdkFocusInitial
+      [routerLink]="['/create-workshop', workshopId]"
+      [mat-dialog-close]="true"
+    >
```
Comment (humanmantis) [inline src/app/shared/components/workshop-card/workshop-seats-lack-modal/workshop-seats-lack-modal.component.html]: Keep the bracket on the same line

File: src/app/shared/components/workshop-card/workshop-card.component.ts [hunk 2]
```diff
@@ -111,29 +116,45 @@ export class WorkshopCardComponent implements OnInit, OnDestroy {
...
-      if (res) {
-        this.store.dispatch(
-          new UpdateWorkshopStatus({ workshopId: this.workshopData.workshopId, status: status }, this.workshopData.providerId)
-        );
-      }
-    });
+      dialogRef.afterClosed().subscribe((res: boolean) => {
+        if (res) {
+          this.store.dispatch(
+            new UpdateWorkshopStatus(
```
Comment (humanmantis) [inline src/app/shared/components/workshop-card/workshop-card.component.ts]: The previous formatting was much more compact

File: src/app/shared/components/workshop-card/workshop-card.component.ts [hunk 3]
```diff
@@ -111,29 +116,45 @@ export class WorkshopCardComponent implements OnInit, OnDestroy {
...
+              {
+                workshopId: this.workshopData.workshopId,
+                status: status
+              },
+              this.workshopData.providerId
+            )
+          );
+        }
+      });
+    } else {
```
Comment (humanmantis) [inline src/app/shared/components/workshop-card/workshop-card.component.ts:134]: Missing spaces in formatting

**Review by humanmantis — CHANGES_REQUESTED** [review]

File: src/app/shared/components/workshop-card/workshop-card.component.ts (hunk 3 above)
Comment (doliinyk) [reply src/app/shared/components/workshop-card/workshop-card.component.ts:134]: Can you specify what spaces are you mean? There are already spaces on these lines, my formatter doesn't change these lines

Comment (humanmantis) [reply src/app/shared/components/workshop-card/workshop-card.component.ts:134]: Check it again. 
Everything is fine, I didn't notice that the parentheses above are closed

**Review by humanmantis — APPROVED** [review]

**Pull request merged**